### PR TITLE
KNOX-2651 - NPE when token value is missing

### PR DESCRIPTION
**Pull request by zeroflag**
## What changes were proposed in this pull request?

There can be a NPE when there is a space between the username and password.

## How was this patch tested?

```
$ curl -vvv -iku Token: xxxx https://localhost:8443/gateway/sandbox/webhdfs/v1/?op=LISTSTATUS
```

### CODE REVIEW

**Comment by zeroflag** [comment]
cc: @pzampino @smolnar82 @moresandeep

File: gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/JWTFederationFilter.java [hunk 1]
```diff
@@ -126,7 +126,7 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
     }
     final Pair<TokenType, String> wireToken = getWireToken(request);
 
-    if (wireToken != null) {
+    if (wireToken != null && wireToken.getLeft() != null && wireToken.getRight() != null) {
```
Comment (smolnar82) [inline gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/JWTFederationFilter.java:129]: I think we could log the fact that token type/value is missing. It'd be a great help when debugging 401 issues.

**Review by smolnar82 — CHANGES_REQUESTED** [review]

File: gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/JWTFederationFilter.java (hunk 1 above)
Comment (zeroflag) [reply gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/JWTFederationFilter.java:129]: Good point.

**Review by pzampino — APPROVED** [review]
LGTM

**Review by smolnar82 — APPROVED** [review]

**Pull request merged**